pantheon/src/main/java/tech/pegasys/pantheon/cli/subcommands/blocks/BlocksSubCommand.java

```diff
@@ -241,71 +241,127 @@ private MiningParameters getMiningParameters() {
-              new CommandLine(this),
-              "Partial export due to inability to recover all requested blocks");
+        switch (format) {
+          case RLP:
+            exportRlpFormat(controller);
+            break;
+          default:
+            throw new ParameterException(
+                spec.commandLine(), "Unsupported format: " + format.toString());
         }
-
+      } catch (IOException e) {
+        throw new ExecutionException(
+            spec.commandLine(), "An error occurred while exporting blocks.", e);
       } finally {
         metricsService.ifPresent(MetricsService::stop);
       }
     }
 
-    private static void checkCommand(
+    private PantheonController<?> createPantheonController() {
+      return parentCommand.parentCommand.buildController();
+    }
+
+    private void exportRlpFormat(final PantheonController<?> controller) throws IOException {
+      final ProtocolContext<?> context = controller.getProtocolContext();
+      RlpBlockExporter exporter =
+          parentCommand.rlpBlockExporterFactory.get(context.getBlockchain());
+      exporter.exportBlocks(blocksExportFile, getStartBlock(), getEndBlock());
+    }
+
+    private void checkCommand(
         final ExportSubCommand exportSubCommand, final Long startBlock, final Long endBlock) {
       checkNotNull(exportSubCommand.parentCommand);
-      checkNotNull(exportSubCommand.parentCommand.blockExporter);
-      checkNotNull(startBlock);
-      if (startBlock < 0) {
-        throw new CommandLine.ParameterException(
-            new CommandLine(exportSubCommand),
-            "--start-block must be greater than or equal to zero");
+
+      final Optional<Long> maybeStartBlock = getStartBlock();
+      final Optional<Long> maybeEndBlock = getEndBlock();
+
+      maybeStartBlock
+          .filter(blockNum -> blockNum < 0)
+          .ifPresent(
+              (blockNum) -> {
+                throw new CommandLine.ParameterException(
+                    spec.commandLine(),
+                    "Parameter --start-block ("
+                        + blockNum
+                        + ") must be greater than or equal to zero.");
+              });
+
+      maybeEndBlock
+          .filter(blockNum -> blockNum < 0)
+          .ifPresent(
+              (blockNum) -> {
+                throw new CommandLine.ParameterException(
+                    spec.commandLine(),
+                    "Parameter --end-block ("
+                        + blockNum
+                        + ") must be greater than or equal to zero.");
+              });
+
+      if (maybeStartBlock.isPresent() && maybeEndBlock.isPresent()) {
+        if (endBlock <= startBlock) {
+          throw new CommandLine.ParameterException(
+              spec.commandLine(),
+              "Parameter --end-block ("
+                  + endBlock
+                  + ") must be greater start block ("
+                  + startBlock
+                  + ").");
+        }
```

Comment:
We don't know the state of the blockchain here correct?  So we can't assert startblock is in the chain?  I think this could be a user error we can detect (somehow) and we should provide the user feedback on.  Perhaps we could count the number of blocks exported and return it as part of the `exportBlocks` method, and log a warning or output the console if no blocks were exported.

Reply:
We'll currently log a warning if any blocks in the defined range were missing.  Might be a good idea to error out if no blocks are exported though ... I may follow-up on that in another PR.